packages/cacti-plugin-consortium-static/src/main/typescript/consortium/process-broadcast-endpoint-v1.ts

```diff
@@ -0,0 +1,125 @@
+import { PluginConsortiumStatic } from "../plugin-consortium-static";
+import { NewConsortiumRepository } from "../repository/new-consortium-repository";
+
+export interface IProcessBroadcastEndpointOptions {
+  plugin: PluginConsortiumStatic;
+  keyPairPem: string;
+  consortiumRepo: NewConsortiumRepository;
+  logLevel?: LogLevelDesc;
+}
+
+export class ProcessBroadcastEndpoint implements IWebServiceEndpoint {
+  public static readonly CLASS_NAME = "ProcessBroadcastEndpoint";
+
+  private readonly log: Logger;
+  private readonly plugin: PluginConsortiumStatic;
+
+  constructor(public readonly options: IProcessBroadcastEndpointOptions) {
+    const fnTag = `${this.className}#constructor()`;
+    if (!options) {
+      throw new Error(`${fnTag} options falsy.`);
+    }
+    if (!options.keyPairPem) {
+      throw new Error(`${fnTag} options.keyPairPem falsy.`);
+    }
+    Checks.truthy(options.consortiumRepo, `${fnTag} options.consortiumRepo`);
+    Checks.truthy(options.plugin, `${fnTag} options.plugin`);
+    Checks.truthy(
+      options.plugin instanceof PluginConsortiumStatic,
+      `${fnTag} options.plugin instanceof PluginConsortiumStatic`,
+    );
+    this.plugin = options.plugin;
+
+    const level = options.logLevel || "INFO";
+    const label = "add-node-endpoint-v1";
+    this.log = LoggerProvider.getOrCreate({ level, label });
+  }
+
+  public get className(): string {
+    return ProcessBroadcastEndpoint.CLASS_NAME;
+  }
+
+  getAuthorizationOptionsProvider(): IAsyncProvider<IEndpointAuthzOptions> {
+    // TODO: make this an injectable dependency in the constructor
+    return {
+      get: async () => ({
+        isProtected: true,
+        requiredRoles: [],
+      }),
+    };
+  }
+
+  public getExpressRequestHandler(): IExpressRequestHandler {
+    return this.handleRequest.bind(this);
+  }
+
+  public get oasPath(): (typeof OAS.paths)["/api/v1/plugins/@hyperledger/cacti-plugin-consortium-static/receive-broadcast"] {
+    return OAS.paths[
+      "/api/v1/plugins/@hyperledger/cacti-plugin-consortium-static/receive-broadcast"
+    ];
+  }
+
+  public getOperationId(): string {
+    return this.oasPath.post.operationId;
+  }
+
+  public getPath(): string {
+    return this.oasPath.post["x-hyperledger-cacti"].http.path;
+  }
+
+  public getVerbLowerCase(): string {
+    return this.oasPath.post["x-hyperledger-cacti"].http.verbLowerCase;
+  }
+
+  public async registerExpress(
+    expressApp: Express,
+  ): Promise<IWebServiceEndpoint> {
+    await registerWebServiceEndpoint(expressApp, this);
+    return this;
+  }
+
+  async handleRequest(req: Request, res: Response): Promise<void> {
+    const fnTag = `${this.className}#handleRequest()`;
+    const reqTag = `${this.getVerbLowerCase()} - ${this.getPath()}`;
+
+    this.log.debug(`GET ${this.getPath()}`);
+    try {
+      await this.plugin.verifySignature(
+        req.body.signature,
+        safeStableStringify(req.body.message) as string,
```

Comment:
@eduv09 Please try to not nest function calls in each other, it makes the stack traces that much harder to read when you see that something crashed on this line but cannot be sure if it was something within `verifySignature` or something within a function that was computing a parameter of `verifySignature`. A separate line for the stringification and a dedicated variable to its output could then solve the issue and it would also allow you to validate the output with somehting like the snippet I'll put below. That would make it so that the casting to string can be removed (which in its current form is an unsafe cast because you are not really sure if its a string or not at runtime).

```typescript
if (typeof msgAsJson !== "string") { throw new InternalServerError("safeStableStringify() returned with non-string value.")}
```